### PR TITLE
[Navigation] Adjusted states and transitions

### PR DESCRIPTION
**Pull request by tjonx**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-summer-editions/issues/748

### WHAT is this pull request doing?

- Changes hover states to lighter color
- Changes some font weights and adds type adjustments
- Adds transition to `background-color`

### CODE REVIEW

**Comment by mateus** [comment]
Changes available here to test https://admin.web.web-eh2p.mateus-ferreira.us.spin.dev/store/shop1/customers

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 1]
```diff
@@ -721,7 +774,8 @@ $nav-max-width: 360px;
 
       // stylelint-disable-next-line selector-max-class, selector-max-combinators, selector-max-compound-selectors -- necessary to override the specificity of the Text class
       .Text {
-        font-weight: var(--p-font-weight-semibold);
+        // stylelint-disable-next-line declaration-no-important -- necessary to override the specificity of the Text class
+        font-weight: var(--p-font-weight-semibold) !important;
```
Comment (sam-b-rose) [inline polaris-react/src/components/Navigation/Navigation.scss:782]: Should we also add a transition for font-weight? Felt a bit odd to me initially seeing the font weight change before other transitions:

transition-speed

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 2]
```diff
@@ -334,6 +365,21 @@ $nav-max-width: 360px;
...
+      content: url('data:image/svg+xml,%3Csvg%20width%3D%2221%22%20height%3D%2228%22%20viewBox%3D%220%200%2021%2028%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%3Cpath%20d%3D%22M9%2024.75C9%2024.3358%209.33579%2024%209.75%2024V24C10.1642%2024%2010.5%2024.3358%2010.5%2024.75V28H9V24.75Z%22%20fill%3D%22%23B5B5B5%22%2F%3E%0A%3C%2Fsvg%3E');
+
+      position: absolute;
+      top: 0;
+      left: var(--p-space-2);
+      width: 21px;
+      height: 28px;
+      border-radius: 0;
+      opacity: 1;
+      transition: opacity var(--p-motion-duration-150) var(--p-motion-ease-out);
```
Comment (sam-b-rose) [inline polaris-react/src/components/Navigation/Navigation.scss:384]: Should we make this transition the same as the background-color or vice versa?

**Review by sam-b-rose — APPROVED** [review]

File: polaris-react/src/components/Navigation/Navigation.scss (hunk 1 above)
Comment (kyledurand) [reply polaris-react/src/components/Navigation/Navigation.scss:782]: The letter spacing change on hover is also throwing me a little here

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 3]
```diff
@@ -199,6 +224,12 @@ $nav-max-width: 360px;
   #{$se23} & {
     background-color: var(--p-color-bg-app-selected);
 
+    // stylelint-disable-next-line selector-max-combinators -- Needed for precision on text
+    .Text {
+      color: var(--p-color-text);
+      letter-spacing: -0.005rem;
+    }
```
Comment (kyledurand) [inline polaris-react/src/components/Navigation/Navigation.scss]: Can we save this number to a localized var that can also be used below?

```suggestion
    .Text {
      --pc-navigation-text-spacing: -0.005rem;
      color: var(--p-color-text);
      letter-spacing: var(--pc-navigation-text-spacing);
    }
```

**Review by kyledurand — APPROVED** [review]
This looks good but should we get some UX 👀 on this for 🎩 as well?

File: polaris-react/src/components/Navigation/Navigation.scss (hunk 1 above)
Comment (tjonx) [reply polaris-react/src/components/Navigation/Navigation.scss:782]: I'd avoid recording and analyzing these transitions in slow motion and at 200% size. Adding transitions on `font-weight` creates some jankiness and the current version of these transitions at normal size and speed look fine.

https://github.com/Shopify/polaris/assets/67433661/0c816aa2-54a0-4dec-a5c4-5c4698fcc31c

**Pull request merged**